cli/azd/pkg/project/framework_service_docker.go

```diff
@@ -180,200 +180,196 @@ func (p *dockerProject) SetSource(inner FrameworkService) {
 func (p *dockerProject) Restore(
 	ctx context.Context,
 	serviceConfig *ServiceConfig,
-) *async.TaskWithProgress[*ServiceRestoreResult, ServiceProgress] {
+	progress *async.Progress[ServiceProgress],
+) (*ServiceRestoreResult, error) {
 	// When the program runs the restore actions for the underlying project (containerapp),
 	// the dependencies are installed locally
-	return p.framework.Restore(ctx, serviceConfig)
+	return p.framework.Restore(ctx, serviceConfig, progress)
 }
 
 // Builds the docker project based on the docker options specified within the Service configuration
 func (p *dockerProject) Build(
 	ctx context.Context,
 	serviceConfig *ServiceConfig,
 	restoreOutput *ServiceRestoreResult,
-) *async.TaskWithProgress[*ServiceBuildResult, ServiceProgress] {
-	return async.RunTaskWithProgress(
-		func(task *async.TaskContextWithProgress[*ServiceBuildResult, ServiceProgress]) {
-			dockerOptions := getDockerOptionsWithDefaults(serviceConfig.Docker)
-
-			resolveParameters := func(source []string) []string {
-				result := make([]string, len(source))
-				for i, arg := range source {
-					evaluatedString, err := apphost.EvalString(arg, func(match string) (string, error) {
-						path := match
-						value, has := p.env.Config.GetString(path)
-						if !has {
-							return "", fmt.Errorf("parameter %s not found", path)
-						}
-						return value, nil
-					})
-					if err != nil {
-						task.SetError(err)
```

Comment:
As part of this refactoring, I discovered this issue where we call `task.SetError` and then return, but that code has been refactored into a helper function (to remove duplication) which closed over the TaskContext. This caused us to just discard this error and likely do bad things.

I suspect this happened as part of a refactoring and is the sort of thing that is harder to do in the new model.